File: 프로그래머스/lv1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    let counter = 0;
    result = '';
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === ' ') {
            result = result + ' ';  
            counter = 0;
        } else {
            if(counter % 2 === 0) {
                result = result + s[i].toUpperCase();
            } else {
                result = result + s[i].toLowerCase();
            }
            counter++;
        }
    }
    return result;
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.py
def solution(s):
    return ' '.join(
        map(lambda word: 
            ''.join(
                [char.upper() if i % 2 == 0 else char.lower() for i, char in enumerate(word)]
            )
        , s.split(' ')))


    # result = []
    # for word in s.split(' '):
    #     converted_word = ''
    #     for i, char in enumerate(word):
    #         if i % 2 == 0:
    #             converted_word += char.upper() 
    #         else: 
    #             converted_word += char.lower()
    #     result.append(converted_word)
    # return ' '.join(result)
            